src/d_soundfile.c

```diff
@@ -495,6 +502,34 @@ static void soundfile_xferin_sample(const t_soundfile *sf, int nvecs,
                 }
             }
         }
+        else if (sf->sf_bytespersample == 8)
+        {
+            t_doubleuint alias;
+            if (sf->sf_bigendian)
+            {
+                for (j = 0, sp2 = sp, fp = vecs[i] + framesread;
+                    j < nframes; j++, sp2 += sf->sf_bytesperframe, fp++)
+                {
+                    alias.ui = (((uint64_t)sp2[0] << 56) | ((uint64_t)sp2[1] << 48) |
+                                ((uint64_t)sp2[2] << 40) | ((uint64_t)sp2[3] << 32) |
+                                ((uint64_t)sp2[4] << 24) | ((uint64_t)sp2[5] << 16) |
+                                ((uint64_t)sp2[6] << 8)  |  (uint64_t)sp2[7]);
+                    *fp = (t_sample)alias.d;
+                }
+            }
+            else
+            {
+                for (j = 0, sp2 = sp, fp = vecs[i] + framesread;
+                    j < nframes; j++, sp2 += sf->sf_bytesperframe, fp++)
+                {
+                    alias.ui = (((uint64_t)sp2[7] << 56) | ((uint64_t)sp2[6] << 48) |
+                                ((uint64_t)sp2[5] << 40) | ((uint64_t)sp2[4] << 32) |
```

Comment:
there's quite a lot of code-duplication here, probably use a macro here and for the reverse operation (also for single)

```C
#define BYTES2DOUBLEUINT(bytes, b0, b1, b2, b3, b4, b5, b6, b7) \
    (((uint64_t)((bytes)[(b0)]) << 56) | ((uint64_t)((bytes)[(b1)]) << 48) | \
     ((uint64_t)((bytes)[(b2)]) << 40) | ((uint64_t)((bytes)[(b3)]) << 32) | \
     ((uint64_t)((bytes)[(b4)]) << 24) | ((uint64_t)((bytes)[(b5)]) << 16) | \
     ((uint64_t)((bytes)[(b6)]) <<  8) | ((uint64_t)((bytes)[(b7)]) <<  0)

            if (sf->sf_bigendian)
            {
                for (j = 0, sp2 = sp, fp = vecs[i] + framesread;
                    j < nframes; j++, sp2 += sf->sf_bytesperframe, fp++)
                {
                    alias.ui = BYTES2DOUBLEUINT(sp2, 0, 1, 2, 3, 4, 5, 6, 7);
                    *fp = (t_sample)alias.d;
                }
            }
            else
            {
                for (j = 0, sp2 = sp, fp = vecs[i] + framesread;
                    j < nframes; j++, sp2 += sf->sf_bytesperframe, fp++)
                {
                    alias.ui = BYTES2DOUBLEUINT(sp2, 7, 6, 5, 4, 3, 2, 1);
```

Reply:
The same is true for the other bytespersamples...

Reply:
(moving this post to the right thread, and hopefully fixing the auto-correction nonsense)

Yes (that's why I wrote "also for single").

Of course the amount of duplicate code is linearly proportional to the bytes/sample, so this is more useful for double precision than e.g. for 16bit PCM.

The idea is mostly to reduce copy/paste errors that are hard to spot on the repetitive sequences